crates/re_renderer/src/renderer/debug_overlay.rs

```diff
@@ -0,0 +1,245 @@
+
+    /// Show the texture on the uint texture binding slot.
+    SHOW_UINT_TEXTURE = 1,
+}
+
+mod gpu_data {
+    use crate::wgpu_buffer_types;
+
+    /// Keep in sync with `debug_overlay.wgsl`
+    #[repr(C, align(256))]
+    #[derive(Clone, Copy, bytemuck::Pod, bytemuck::Zeroable)]
+    pub struct DebugOverlayUniformBuffer {
+        pub screen_resolution: wgpu_buffer_types::Vec2,
+        pub position_in_pixel: wgpu_buffer_types::Vec2,
+        pub extent_in_pixel: wgpu_buffer_types::Vec2,
+        pub mode: u32,
+        pub _padding: u32,
+        pub end_padding: [wgpu_buffer_types::PaddingRow; 16 - 2],
+    }
+}
+
+pub struct DebugOverlayRenderer {
+    render_pipeline: GpuRenderPipelineHandle,
+    bind_group_layout: GpuBindGroupLayoutHandle,
+}
+
+/// Debug overlay for quick & dirty display of texture contents.
+///
+/// Executed as part of the composition draw phase in order to allow "direct" output to the screen.
+///
+/// Do *not* use this in production!
+/// See also `debug_overlay.wgsl` - you are encouraged to edit this shader for your concrete debugging needs!
+#[derive(Clone)]
+pub struct DebugOverlayDrawData {
+    bind_group: GpuBindGroup,
+}
+
+impl DrawData for DebugOverlayDrawData {
+    type Renderer = DebugOverlayRenderer;
+}
+
+impl DebugOverlayDrawData {
+    pub fn new(
+        ctx: &mut RenderContext,
+        debug_texture: &GpuTexture,
+        screen_resolution: glam::UVec2,
+        position_in_pixel: glam::IVec2,
+        extent_in_pixel: glam::UVec2,
+    ) -> Self {
+        let mut renderers = ctx.renderers.write();
+        let debug_overlay = renderers.get_or_create::<_, DebugOverlayRenderer>(
+            &ctx.shared_renderer_data,
+            &mut ctx.gpu_resources,
+            &ctx.device,
+            &mut ctx.resolver,
+        );
+
+        let mode = match debug_texture.texture.format().describe().sample_type {
+            wgpu::TextureSampleType::Depth | wgpu::TextureSampleType::Float { filterable: _ } => {
+                DebugOverlayMode::SHOW_FLOAT_TEXTURE
+            }
+            wgpu::TextureSampleType::Sint | wgpu::TextureSampleType::Uint => {
+                DebugOverlayMode::SHOW_UINT_TEXTURE
+            }
+        };
+
+        let uniform_buffer_binding = create_and_fill_uniform_buffer(
+            ctx,
+            "DebugOverlayDrawData".into(),
+            gpu_data::DebugOverlayUniformBuffer {
+                screen_resolution: screen_resolution.as_vec2().into(),
+                position_in_pixel: position_in_pixel.as_vec2().into(),
+                extent_in_pixel: extent_in_pixel.as_vec2().into(),
+                mode: mode as u32,
+                _padding: 0,
+                end_padding: Default::default(),
+            },
+        );
+
+        let fallback_texture = ctx
+            .texture_manager_2d
+            .get(ctx.texture_manager_2d.white_texture_handle())
+            .expect("white fallback texture missing")
+            .handle;
```

Comment:
The same texture can be used for both the int and float fallback texture?

Reply:
for some reason I thought I had this tried and it worked surprisingly. Doubted that now and no it doesn't. Will introduce a integer version of the fallback texture